tests/integration/test_broadcast.py

```diff
@@ -0,0 +1,200 @@
+# Copyright 2022 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import numpy as np
+import pytest
+
+import cunumeric as num
+from legate.core import LEGATE_MAX_DIM
+
+DIM_CASES = [5, 40]
+
+
+def _print_result(test_result, print_msg, err_arr):
+    assert test_result, (
+        f"Failed, {print_msg}\n"
+        f"Attr, {err_arr[0]}\n"
+        f"numpy result: {err_arr[1]}\n"
+        f"cunumeric_result: {err_arr[2]}\n"
+        f"cunumeric and numpy shows"
+        f" different result\n"
+    )
+    print(f"Passed, {print_msg}")
+
+
+def _broadcast_attrs(sizes):
+    arr_np = list(np.arange(np.prod(size)).reshape(size) for size in sizes)
+    arr_num = list(num.arange(np.prod(size)).reshape(size) for size in sizes)
+    b = np.broadcast(*arr_np)
+    c = num.broadcast(*arr_num)
+
+    attrs = ["index", "nd", "ndim", "numiter", "shape", "size"]
+    is_equal = True
+    err_arr = None
+    # test attributes
+    for attr in attrs:
+        if getattr(b, attr) != getattr(c, attr):
+            is_equal = False
+            err_arr = [attr, getattr(b, attr), getattr(c, attr)]
```

Comment:
Here and elsewhere, `is_equal` seems redundant, since `err_arr` is None if and only `is_equal` is True.  Another approach here would be to initialize `err_arr` to be an empty list, and then instead of breaking on the first mismathc, collect all errors. Then after the loop things were equal if `err_arr` is still empty, and if not, you can print out more diagnostic information (all the mismatches, not just the first)

Reply:
Done.